Improve macro-op fusion: skip nop and add lui + addi

qwe661234 opened the pull request:
1. Refine origin fused instruction by skipping insturction nop and correctly updating value to register.
2. Add new fused insturction lui + addi.

Benchmark dhrystone gains about 3% performance improvement base on this modification.

Close: #177

jserv commented on src/emulate.c:
```diff
@@ -312,15 +312,7 @@ static uint32_t last_pc = 0;
-    const rv_insn_t *next = ir + 1;
-    MUST_TAIL return next->impl(rv, next);
-}
+RVOP(nop, {/* no operation */})
```
Why did you change `nop`?

jserv commented:
Are you ready to close #177 ?

Commit:
Refine instruction fusion and add new one

1. Refine origin fused instruction by skipping insturction nop and
   correctly updating value to register.
2. Add new fused insturction lui + addi.

Benchmark dhrystone gains about 3% performance improvement base on this
modification.

Close: #177

qwe661234 commented:
> Are you ready to close #177 ?

Yes, we cannot find `do_nop` in perf report after this modification.
